src/entities/tag.cpp

```diff
@@ -712,23 +711,26 @@ QVector<int> Tag::fetchAllLinkedNoteIds(int tagId, const bool fromAllSubfolders,
     if (!query.exec()) {
         qWarning() << __func__ << ": " << query.lastError();
     } else {
+        QVector<int> noteIdList;
         for (int r = 0; query.next(); r++) {
             // always keep in mind that note_file_name is no file name,
             // but the base name (so "my-note", instead of "my-note.md")
             const QString &name =
                 query.value(QStringLiteral("note_file_name")).toString();
             const QString &noteSubFolderPathData =
                 query.value(QStringLiteral("note_sub_folder_path")).toString();
-            const Note &note = Note::fetchByName(name, noteSubFolderPathData,
-                                                 QStringLiteral("/"));
-
-            noteIdList.append(note.getId());
+            int noteSubFolderId =
+                NoteSubFolder::fetchByPathData(noteSubFolderPathData, QStringLiteral("/"))
+                    .getId();
+            int noteId = Note::fetchNoteIdByName(name, noteSubFolderId);
+            noteIdList.append(noteId);
         }
+        return noteIdList;
```

Comment:
If you return here then `DatabaseService::closeDatabaseConnection` will not be called!

Reply:
I'll fix that.

Reply:
@pbek does it really matter? The db is local and is always accessed from a single thread

Reply:
It "should" not matter, but it did on Windows. 😬

Reply:
see: `DatabaseService::closeDatabaseConnection`